src/components/posts/PostsPage.js

```diff
@@ -4,77 +4,76 @@ import PostForm from './PostForm';
-  const [posts, setPosts] = useState([]);
+	const [post, setPost] = useState({ content: '' });
+	const [postResponse, setPostResponse] = useState('');
+	const [posts, setPosts] = useState([]);
+	
+	useEffect(() => {
+		client.get('/posts').then((res) => {
+			setPosts(res.data.data.posts);
+		});
+	}, [postResponse]);
 
-  useEffect(() => {
-    client.get('/posts').then((res) => {
-      setPosts(res.data.data.posts);
-    });
-  }, [postResponse]);
+	const createPost = (event) => {
+		event.preventDefault();
+		client
+			.post('/post', post)
+			.then((res) => {
+				setPostResponse(res.data);
+				setPosts((posts) => [res.data.data.post, ...posts]);
+			})
+			.catch((err) => {
+				console.log(err.message, 'Invalid Post');
+			});
+		setPost(() => ({ content: '' }));
+	};
 
-  const createPost = (event) => {
-    event.preventDefault();
-    client
-      .post('/post', post)
-      .then((res) => {
-        setPostResponse(res.data);
-        setPosts((posts) => [res.data.data.post, ...posts]);
-      })
-      .catch((err) => {
-        console.log(err.message, 'Invalid Post');
-      });
-    setPost(() => ({ content: '' }));
-  };
+	const onCommentAdded = (post, comment) => {
+		setPosts(
+			posts.map((updatedPost) =>
+				updatedPost === post
+					? { ...post, postComments: [comment, ...post.postComments] }
+					: updatedPost
+			)
+		);
+	};
 
-  const onCommentAdded = (post, comment) => {
-    setPosts(
-      posts.map((updatedPost) =>
-        updatedPost === post
-          ? { ...post, postComments: [comment, ...post.postComments] }
-          : updatedPost
-      )
-    );
-  };
+	const handleChange = (event) => {
+		event.preventDefault();
+		const { value, name } = event.target;
+		setPost({
+			...post,
+			[name]: value,
+		});
+	};
 
-  const handleChange = (event) => {
-    event.preventDefault();
-    const { value, name } = event.target;
-    setPost({
-      ...post,
-      [name]: value,
-    });
-  };
-
-  return (
-    <>
-    <div className='home-container'>
-    <section className='posts-section'>
-      <div>
-        {postResponse.status}
-        <PostForm
-          handleSubmit={createPost}
-          handleChange={handleChange}
-          inputValue={post.content}
-        />
-        <ul className='posts-list'>
-          {posts.map((post, index) => (
-            <li key={index} className='post-item'>
-              <Post post={post} onCommentAdded={onCommentAdded} />
-            </li>
-          ))}
-        </ul>
-        </div>
-      </section>   
-    {localStorage.getItem('role') === 'TEACHER' && <CohortList />}   
-    </div> 
-  </>
-  );
+	return (
+		<>
+			<div className='home-container'>
+				<section className='posts-section'>
+					<div>
+						{postResponse.status}
+						<PostForm
+							handleSubmit={createPost}
+							handleChange={handleChange}
+							inputValue={post.content}
+						/>
+						<ul className='posts-list'>
+							{posts.map((post, index) => (
```

Comment:
I think you should refactor this mapping into another component. This will make the JSX easier to read

Reply:
this was not actually our code. for some reason when I pulled from main and tried to merge the changes I had 9 conflicts that I had to resolve and this has caused our changes to be bigger than they are.